src/rdkafka_cgrp.c

```diff
@@ -3285,6 +3321,21 @@ void rd_kafka_cgrp_handle_ConsumerGroupHeartbeat(rd_kafka_t *rk,
                     rkcg, "coordinator query");
         }
 
+        if (actions & RD_KAFKA_ERR_ACTION_SPECIAL &&
+            rkcg->rkcg_flags & RD_KAFKA_CGRP_F_SUBSCRIPTION) {
```

Comment:
- We send HB only if we have this flag set, so we can skip this check. There might be a case where we unsubscribed and then sending this error once. I think that should be fine. If not, then we can add this check with error check case.
- This is also more specific to TopicAuthorizationError.
- This won't be correct if we add a new Error here.

Reply:
> This is also more specific to TopicAuthorizationError.

In case we'll have more errors that set this flag and we need to do something different we can use a switch on the error code

Reply:
Add a comment.

Reply:
ok removed the condition as in that case the error in the response is converted to `OUTDATED`